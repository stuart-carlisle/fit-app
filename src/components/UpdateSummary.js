import React from 'react'
import MultiColorProgressBar from './MultiColorProgressBar'

export const UpdateSummary =({data}) => {
    const calorieTarget = data.targets.calorieTarget
    const carbsTarget = data.targets.carbsTarget
    const fatTarget = data.targets.fatTarget

    const calories = data.calories
    const fat = data.fat
    const carbs = data.carbs

    const percentageCalories = Math.round((calories/ calorieTarget)*100)
    const percentageCaloriesLeft = percentageCalories ? 100-percentageCalories : 100
    
    const readingsCalories = percentageCaloriesLeft <= 0 ? 
    [
        {
            value: 100,
            color: '#ff0808'
        }
    ] :
    
    [
        {
            value: percentageCalories,
            color: '#4472C7'
        },
        {
            value: percentageCaloriesLeft >= 100 ? 100 : percentageCaloriesLeft,
            color: '#48e880'
        }
    ]
  
    const percentageCarbs = Math.round((carbs/ carbsTarget)*100)
    const percentageCarbsLeft = percentageCarbs ? 100-percentageCarbs : 100
    
    const readingsCarbs = percentageCarbsLeft <= 0 ? 
    [
        {
            value: 100,
            color: '#ff0808'
        }
    ] :
    
    [
        {
            value: percentageCarbs,
            color: '#4472C7'
        },
        {
            value: percentageCarbsLeft >= 100 ? 100 : percentageCarbsLeft,
            color: '#48e880'
        }
    ]

    const percentageFat = Math.round((fat/ fatTarget)*100)
    const percentageFatLeft = percentageFat ? 100-percentageFat : 100
    
    const readingsFat = percentageFatLeft <= 0 ? 
    [
        {
            value: 100,
            color: '#ff0808'
        }
    ] :
    
    [
        {
            value: percentageFat,
            color: '#4472C7'
        },
        {
            value: percentageFatLeft >= 100 ? 100 : percentageFatLeft,
            color: '#48e880'
        }
    ]
   
      return (
        <div>
        <span className="text-input__span-container2">
            <div className="update-summary_item">
                <h3 className="form__heading__update-summary">{`${calories} Cal`}</h3>
                <MultiColorProgressBar key={1} readings={readingsCalories}/>
            </div>
            <div className="update-summary_item">
                <h3 className="form__heading__update-summary">{`${carbs}g Carbs`}</h3>
                <MultiColorProgressBar key={2} readings={readingsCarbs}/>
            </div>
            <div className="update-summary_item">
                <h3 className="form__heading__update-summary">{`${fat}g Fats`}</h3>
                <MultiColorProgressBar key={3} readings={readingsFat}/>
            </div> 
            </span>
        </div>
      );
}
  
  
export default UpdateSummary