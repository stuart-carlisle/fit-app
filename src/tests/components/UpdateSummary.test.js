import React from 'react'
import { shallow } from 'enzyme'
import UpdateSummary from '../../components/UpdateSummary'

test('Should render the update summary component correctly',()=>{
    const data={
        targets:{
            calorieTarget:2000,
            carbsTarget:49,
            fatTarget:3
        },
        calories:1400,
        fat:2,
        carbs:37
    }
    const wrapper = shallow(<UpdateSummary data={data} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render the update summary component correctly with data greater than targets',()=>{
    const data={
        targets:{
            calorieTarget:2000,
            carbsTarget:49,
            fatTarget:3
        },
        calories:2400,
        fat:4,
        carbs:56
    }
    const wrapper = shallow(<UpdateSummary data={data} />)
    expect(wrapper).toMatchSnapshot()
})

