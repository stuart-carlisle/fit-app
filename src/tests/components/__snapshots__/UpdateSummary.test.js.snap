// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the update summary component correctly 1`] = `
<div>
  <span
    className="text-input__span-container2"
  >
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        1400 Cal
      </h3>
      <MultiColorProgressBar
        key="1"
        readings={
          Array [
            Object {
              "color": "#4472C7",
              "value": 70,
            },
            Object {
              "color": "#48e880",
              "value": 30,
            },
          ]
        }
      />
    </div>
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        37g Carbs
      </h3>
      <MultiColorProgressBar
        key="2"
        readings={
          Array [
            Object {
              "color": "#4472C7",
              "value": 76,
            },
            Object {
              "color": "#48e880",
              "value": 24,
            },
          ]
        }
      />
    </div>
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        2g Fats
      </h3>
      <MultiColorProgressBar
        key="3"
        readings={
          Array [
            Object {
              "color": "#4472C7",
              "value": 67,
            },
            Object {
              "color": "#48e880",
              "value": 33,
            },
          ]
        }
      />
    </div>
  </span>
</div>
`;

exports[`Should render the update summary component correctly with data greater than targets 1`] = `
<div>
  <span
    className="text-input__span-container2"
  >
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        2400 Cal
      </h3>
      <MultiColorProgressBar
        key="1"
        readings={
          Array [
            Object {
              "color": "#ff0808",
              "value": 100,
            },
          ]
        }
      />
    </div>
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        56g Carbs
      </h3>
      <MultiColorProgressBar
        key="2"
        readings={
          Array [
            Object {
              "color": "#ff0808",
              "value": 100,
            },
          ]
        }
      />
    </div>
    <div
      className="update-summary_item"
    >
      <h3
        className="form__heading__update-summary"
      >
        4g Fats
      </h3>
      <MultiColorProgressBar
        key="3"
        readings={
          Array [
            Object {
              "color": "#ff0808",
              "value": 100,
            },
          ]
        }
      />
    </div>
  </span>
</div>
`;
